src/Key.ts

```diff
@@ -0,0 +1,147 @@
+const isMac = navigator.platform === 'MacIntel';
+
+/**
+ * a
+ */
+interface KeyModifierOptions {
+  /** a */
+  alt?: boolean;
+  /** a */
+  ctrl?: boolean;
+  /** a */
+  shift?: boolean;
+}
+
+const codeMap = new Map(
+  Object.entries({
+    Comma: ',',
+    Period: '.',
+    Semicolon: ';',
+    Quote: '"',
+    BracketLeft: '[',
+    BracketRight: ']',
+    Backquote: '`',
+    Backslash: '\\',
+    Minus: '-',
+    Equal: '='
+  })
+);
+
+const keyCodeMap = new Map<number, string>([
+  [37, 'ArrowLeft'],
+  [38, 'ArrowUp'],
+  [39, 'ArrowRight'],
+  [40, 'ArrowDown']
+]);
+
+/**
+ * Code comes in many formats.
+ * Keys start with Key (ex: KeyN)
+ * Digits start with Digit (ex: Digit1)
+ * Other characters like .,[] are all treated with special string values.
+ */
+function getCodeValue(code: string): string | undefined {
+  if (code.startsWith('Key')) {
+    return code.substring(3, code.length).toLowerCase();
+  }
+
+  if (code.startsWith('Digit')) {
+    return code.substring(5, code.length).toLowerCase();
+  }
+
+  return codeMap.get(code);
+}
+
+/**
+ * Gets the key code value.
+ */
+function getKeyCodeValue(keyCode: number): string | undefined {
+  if (keyCodeMap.has(keyCode)) {
+    return keyCodeMap.get(keyCode);
+  }
+
+  return String.fromCharCode(keyCode).toLowerCase();
+}
+
+/**
+ * https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/keyCode
+ * Value detection is hard. Basically we want to use code if we can but code comes in many formats.
+ * Also code is not garuntueed in all browsers so we need to fallback to keyCode or key.
+ */
+function getKeyValue(event: KeyboardEvent): string {
+  if (event.code) {
+    const value = getCodeValue(event.code);
+    if (value) {
+      return value;
+    }
+  }
+
+  if (event.keyCode) {
+    const keyCodeValue = getKeyCodeValue(event.keyCode);
+    if (keyCodeValue) {
+      return keyCodeValue;
+    }
+  }
+
+  return event.key;
+}
+
+/**
+ * Key class for converting and handling key events.
+ */
+class Key {
```

Comment:
Honestly this could probably be a module itself but I do not think we want to break this out and spread that dependency around.